### PR TITLE
open eip2028 when the version of Istanbul was enabled

### LINKED ISSUE #730: The param 'isEIP2028' of function IntrinsicGas should be true if Istanbul version enabled
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

__Steps to reproduce:__
the param 'isEIP2028' of function `IntrinsicGas` which was called in `TransitionDb` Need reset false to true cause Istanbul version has already be enabled
```go
func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (*ExecutionResult, error) {
	contractCreation := st.Recipient == nil
 
       //here need reset false to true cause Istanbul version has already be enabled
	cost, err := core.IntrinsicGas(st.Payload, contractCreation,true,false)
	if err != nil {
		return nil, sdkerrors.Wrap(err, "invalid intrinsic gas for transaction")
	}

	// This gas limit the the transaction gas limit with intrinsic gas subtracted
	gasLimit := st.GasLimit - ctx.GasMeter().GasConsumed()
```
__Expected behavior:__ [What you expected to happen]

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

### PR DESCRIPTION
**Pull request by KamiD**
Closes: #730 

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/chain_config.go [hunk 1]
```diff
@@ -66,6 +66,16 @@ func (cc ChainConfig) EthereumConfig(chainID *big.Int) *params.ChainConfig {
 	}
 }
 
+// IsIstanbul returns whether the Istanbul version enabled.
+func (cc ChainConfig) IsIstanbul() bool {
+	return getBlockValue(cc.IstanbulBlock) != nil
+}
+
+// IsIstanbul returns whether the Istanbul version enabled.
```
Comment (fedekunze) [inline x/evm/types/chain_config.go]:
```suggestion
// IsHomestead returns whether the Homestead version is enabled.
```

File: x/evm/types/chain_config.go [hunk 2]
```diff
@@ -66,6 +66,16 @@ func (cc ChainConfig) EthereumConfig(chainID *big.Int) *params.ChainConfig {
 	}
 }
 
+// IsIstanbul returns whether the Istanbul version enabled.
```
Comment (fedekunze) [inline x/evm/types/chain_config.go]:
```suggestion
// IsIstanbul returns whether the Istanbul version is enabled.
```

**Review by fedekunze — APPROVED** [review]
LGTM, minor comment

**Comment by KamiD** [comment]
> LGTM, minor comment

Cool, all comments has been accepted

**Comment by fedekunze** [comment]
the RPC test needs to be fixed, then I can merge it

**Comment by KamiD** [comment]
> the RPC test needs to be fixed, then I can merge it

RPC test has been fixed

**Pull request merged**